tailormap-components/projects/core/src/lib/user-interface/attributelist/attributelist-table/attributelist-table.component.ts

```diff
@@ -269,19 +279,25 @@ export class AttributelistTableComponent implements AttributelistTable, OnInit,
     const filterForFeatureTypes = new Map<number, string>();
     let filter = '';
     const relatedFeatures = [];
-    const checkedFeatures = this.dataSource.getCheckedRowsAsAttributeListFeature();
+    let checkedFeatures = this.dataSource.getCheckedRowsAsAttributeListFeature();
     checkedFeatures.forEach((row) => {
       const related = row.related_featuretypes;
       related.forEach((r) => {
         if (filterForFeatureTypes.has(r.id)) {
           filter = filterForFeatureTypes.get(r.id);
-          filterForFeatureTypes.set(r.id, filter += ' OR ' + r.filter );
+          const value = r.filter.split('=')[1].split(' ')[1];
+          filterForFeatureTypes.set(r.id, filter += ', ' + value );
         } else {
-          filterForFeatureTypes.set(r.id, r.filter);
+          const column = r.filter.split('=')[0].split(' ')[0];
+          const value = r.filter.split('=')[1].split(' ')[1];
+          filterForFeatureTypes.set(r.id, column + ' IN (' + value);
```

Comment:
Ik zou dit soort functionaliteit om filters te maken in een helper class stoppen